File: LeetCode/Easy/0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
import java.util.Stack;


class MyQueue {

        Stack<Integer> mainStack;
        Stack<Integer> supportStack;

        public MyQueue() {
            mainStack = new Stack<>();
            supportStack = new Stack<>();
        }

        public void push(int x) {

            mainStack.push(x);
        }

        public int pop() {
            
            if(supportStack.isEmpty()) {
                while(!mainStack.isEmpty()) {
                    supportStack.push(mainStack.pop());
                }
            }
            return supportStack.pop();
        }

        public int peek() {
            if(supportStack.isEmpty()) {
                while(!mainStack.isEmpty()) {
                    supportStack.push(mainStack.pop());
                }
            }
            return supportStack.peek();
        }

        public boolean empty() {
            return mainStack.isEmpty() && supportStack.isEmpty();
        }
    }

File: LeetCode/Easy/0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.main_stack = []
        self.support_stack = []
        

    def push(self, x: int) -> None:
        self.main_stack.append(x)

    def pop(self) -> int:
        if len(self.support_stack)==0:
            while not len(self.main_stack)==0:
                self.support_stack.append(self.main_stack.pop())
        return self.support_stack.pop()
        
    def peek(self) -> int:
        if len(self.support_stack)==0:
                    while not len(self.main_stack)==0:
                        self.support_stack.append(self.main_stack.pop())
        return self.support_stack[-1]
        

    def empty(self) -> bool:
         return len(self.support_stack) == 0 and len(self.main_stack) == 0